tool.go

```diff
@@ -0,0 +1,113 @@
+package codacytool
+
+import (
+	"encoding/json"
+	logrus "github.com/sirupsen/logrus"
+	"os"
+	"strings"
+)
+
+// ToolImplementation interface to implement the tool
+type ToolImplementation interface {
+	Run(tool Tool, sourceDir string) ([]Issue, error)
+}
+
+// Tool represents a codacy tool
+type Tool struct {
+	Definition ToolDefinition
+	config     Configuration
+	Patterns   []Pattern
+	Files      []string
+}
+
+func patternsFromConfig(toolName string, config Configuration) []Pattern {
+	for _, t := range config.Tools {
+		if t.Name == toolName {
+			return t.Patterns
+		}
+	}
+	return []Pattern{}
+}
+
+func newTool(toolDefinition ToolDefinition, config Configuration) Tool {
+	return Tool{
+		Definition: toolDefinition,
+		config:     config,
+		Files:      config.Files,
+		Patterns:   patternsFromConfig(toolDefinition.Name, config),
+	}
+}
+
+func defaultTool(runConfig RunConfiguration) Tool {
+	toolDefinition, err := LoadToolDefinition(defaultDefinitionFile(runConfig.toolConfigsBasePath))
+	if err != nil {
+		panic(err)
+	}
+	config, err := ParseConfiguration(defaultConfigurationFile(runConfig.toolConfigsBasePath))
+	if err != nil {
+		logrus.Debug(defaultConfigurationFile(runConfig.toolConfigsBasePath) + " parsing error: " + err.Error())
+	}
+
+	return newTool(toolDefinition, config)
+}
+
+func appendResult(currentResultString, newResult string) string {
+	return currentResultString + newResult + "\n"
```

Comment:
WDYT about instead of directly concatenating strings, we collect `[]string` and at the end we print it with the proper line ending etc.?

Reply:
Nice suggestion 🙏 got a lot better!